core/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java

```diff
@@ -361,12 +359,19 @@ public Entry snapshot(final Snapshot snapshot) {
     }
 
     @Override
-    public String type() {
+    public String getWriteableName() {
         return TYPE;
     }
 
-    @Override
-    public SnapshotsInProgress readFrom(StreamInput in) throws IOException {
+    public static SnapshotsInProgress readFrom(StreamInput in) throws IOException {
+        return new SnapshotsInProgress(in);
+    }
+
+    public static NamedDiff<Custom> readDiffFrom(StreamInput in) throws IOException {
```

Comment:
I didn't move readFrom to constructor to reduce the number of changes. I will do that. I wouldn't want to move code in readDiffFrom away from the class itself, because it will expose too much implementation details in the registration code.